aten/src/ATen/function_wrapper.py

```diff
@@ -533,7 +537,7 @@ def add_type_as_dynamic_type(argument, option):
 
         # ensure we get reference-type formals when appropriate
         def native_translate_formals(argument, option):
-            if option['inplace']:
+            if (option['inplace'] and argument['name'] == 'self') or argument.get('output', False):
```

Comment:
Is this exercised at all today?

Reply:
No, @colesbury need the `self` check for another PR and the `output` check is from when we were going to implement this as an _out variant.  I left them because they are both technically correct.

Reply:
OK cool; that is what I thought but it is confusing to see non-exercised code (esp. because our codegen is hyper specialized for what declarations we actually have ;)